### PR TITLE
Update REST API after alert refactoring.

### PR DESCRIPTION
**Pull request by jana-rapava**
This PR is part of alert refactoring. Travis fails because it cannot find std::regex in a dependency.

### CODE REVIEW

File: src/web/src/alert_ack.ecpp [hunk 1]
```diff
@@ -67,205 +79,174 @@ bool database_ready;
...
-        std::stringstream input (request.getBody (), std::ios_base::in);
-        cxxtools::JsonDeserializer deserializer (input);
-        deserializer.deserialize (si);
-        si.getMember ("state") >>= state;
-        if (state.empty ())
-            throw std::runtime_error (TRANSLATE_ME ("property 'state' has empty value"));
-    }
-    catch (const std::exception& e) {
-        log_debug ("Bad request document - invalid json: %s", e.what ());
-        std::string err = TRANSLATE_ME ("Please check RFC-11 for valid json schema description.");
```
Comment (jeffdegott) [inline src/web/src/alert_ack.ecpp]: shall be "Invalid json schema description" (avoid reference to RFC-11)

File: src/web/src/alert_ack.ecpp [hunk 2]
```diff
@@ -67,205 +79,174 @@ bool database_ready;
...
+            std::stringstream input (request.getBody (), std::ios_base::in);
+            cxxtools::JsonDeserializer deserializer (input);
+            deserializer.deserialize (si);
+            si.getMember ("state") >>= state;
+            if (state.empty ())
+                throw std::runtime_error (TRANSLATE_ME ("property 'state' has empty value"));
+        }
+        catch (const std::exception& e) {
+            log_debug ("Bad request document - invalid json: %s", e.what ());
+            std::string err = TRANSLATE_ME ("Expected format: { \"state\" : <state> }");
```
Comment (jana-rapava) [inline src/web/src/alert_ack.ecpp]: @jeffdegott , You are commenting part of code which was deleted, new message is here :-)

File: src/web/src/alert_ack.ecpp [hunk 3]
```diff
@@ -67,205 +79,174 @@ bool database_ready;
...
-        log_error ("Alert identified by rule_name = '%s' and element_name = '%s' does not exist.",
-            checked_rule_name.c_str (), asset_id.c_str ());
-        std::string msg_not_found = TRANSLATE_ME ("Alert identified by rule name = '%s' and element name = '%s'", checked_rule_name.c_str (), checked_element_name.c_str ());
-        http_die ("not-found", msg_not_found.c_str ());
+
+    std::string json;
+    ZstrGuard part (zmsg_popstr (resp));
+    if (streq (part, "CHANGESTATE")) {
+        part = zmsg_popstr (resp);
+        if (!part || !streq (part, zuuid_str_canonical (uuid))) {
```
Comment (jeffdegott) [inline src/web/src/alert_ack.ecpp]: does this test make sense?
uuid was destroyed at line 174

File: src/web/src/alert_list.ecpp [hunk 4]
```diff
@@ -164,249 +213,130 @@ bool database_ready;
...
     }
-    if (streq (part, "NOT_FOUND")) {
-        log_error ("State = '%s' does not exist.", checked_state.c_str ());
-        std::string msg = TRANSLATE_ME ("State '%s'", checked_state.c_str ());
-        http_die ("not-found", msg.c_str ());
+
+    std::string json;
+    ZstrGuard part (zmsg_popstr (resp));
+    if (streq (part, "LISTALL")) {
+        part = zmsg_popstr (resp);
```
Comment (jeffdegott) [inline src/web/src/alert_list.ecpp]: should add a comment as "first part ignored"

File: src/web/src/alert_ack.ecpp (hunk 3 above)
Comment (jana-rapava) [reply src/web/src/alert_ack.ecpp]: Good catch, I've forgotten to merge changes from helper branch.

File: src/web/src/alert_list.ecpp [hunk 5]
```diff
@@ -75,18 +70,73 @@ is_requeststate (const char *state) {
...
 
-static int
-is_state_included (const char *request, const char *alert) {
-    if (!is_requeststate (request))
-        return 0;
-    if (!is_alertstate (alert))
-        return 0;
+static zmsg_t*
+s_request_list (zuuid_t *uuid, const std::string& filter, const std::vector<std::string>& desired_elements) {
+    zmsg_t *send_msg = zmsg_new ();
```
Comment (jeffdegott) [inline src/web/src/alert_list.ecpp]: shall check if uuid arg is NULL.

File: src/web/src/alert_list.ecpp [hunk 6]
```diff
@@ -164,249 +213,130 @@ bool database_ready;
...
 
-// send it
-if (mlm_client_sendto (client, AGENT_FTY_ALERT_LIST, RFC_ALERTS_LIST, NULL, 1000, &send_msg) != 0) {
-    log_debug ("mlm_client_sendto (address = '%s', subject = '%s', tracker = NULL, timeout = '%" PRIu32"') failed.",
-        AGENT_FTY_ALERT_LIST, RFC_ALERTS_LIST, 1000);
-    zmsg_destroy (&send_msg);
-    std::string err =  TRANSLATE_ME ("mlm_client_sendto() failed.");
-    http_die ("internal-error", err.c_str ());
-}
+    zuuid_t *uuid = zuuid_new ();
```
Comment (jeffdegott) [inline src/web/src/alert_list.ecpp]: ZuuidGuard usage?

File: src/web/src/my_profile.ecpp [hunk 7]
```diff
@@ -71,6 +71,7 @@
               //Nothing to do, just keep the default value.
             }
           }
+          zconfig_destroy (&root);
```
Comment (jeffdegott) [inline src/web/src/my_profile.ecpp]: useless zconfig_destroy ()

File: src/web/src/alert_ack.ecpp (hunk 2 above)
Comment (jeffdegott) [reply src/web/src/alert_ack.ecpp]: uuuh ;)

**Comment by jana-rapava** [comment]
@jeffdegott , thanks for review :-) I responded to your first comment and updated the PR.

File: packaging/debian/control [hunk 8]
```diff
@@ -119,24 +119,3 @@ Depends:
 Description: fty-rest debugging symbols
  This package contains the debugging symbols for fty-rest:
  common core rest api for 42ity project.
-
```
Comment (perrettecl) [inline packaging/debian/control]: This part must remain

**Pull request merged**